go/workflow.go

```diff
@@ -0,0 +1,270 @@
+
+		if _, ok := err.(*serviceerror.WorkflowExecutionAlreadyStarted); ok {
+			emailToSend = common.EmailGuestInvited
+			previousTier := customer.Status.PreviousTier()
+			err = guestWorkflow.SignalChildWorkflow(ctx, common.SignalEnsureMinimumStatus, previousTier).
+				Get(ctx, nil)
+			if _, ok := err.(*serviceerror.WorkflowExecutionAlreadyStarted); ok {
+				logger.Info("Failed to signal 'already started' guest account; child workflow likely closed.")
+				emailToSend = common.EmailGuestCanceled
+			} else if err != nil {
+				return errors.Wrapf(err, "Could not signal guest/child workflow for guest Id '%v'.", guestId)
+			}
+		} else if err != nil {
+			return errors.Wrapf(err, "Could not start guest/child workflow for guest Id '%v'.", guestId)
+		}
+	} else {
+		logger.Info("Customer does not have sufficient status to invite more guests.")
+		emailToSend = common.EmailInsufficientPoints
+	}
+
+	err = workflow.ExecuteActivity(ctx, activities.SendEmail, emailToSend).Get(ctx, nil)
+	if err != nil {
+		return errors.Wrap(err, "Error running SendEmail activity.")
+	}
+
+	return nil
+}
+
+func signalEnsureMinimumStatus(ctx workflow.Context, c workflow.ReceiveChannel, customer *CustomerInfo) (err error) {
+	logger := workflow.GetLogger(ctx)
+	activities := Activities{}
+
+	var minStatus status.Tier
+	c.Receive(ctx, &minStatus)
+
+	promoted := customer.Status.EnsureMinimum(minStatus)
+
+	if promoted {
+		emailBody := fmt.Sprintf(common.EmailPromoted, minStatus.Name)
+		err := workflow.ExecuteActivity(ctx, activities.SendEmail, emailBody).Get(ctx, nil)
+		if err != nil {
+			logger.Error("Error running SendEmail activity.", "Error", err)
+			return err
+		}
+	}
+
+	return nil
+}
+
+func signalCancelAccount(ctx workflow.Context, c workflow.ReceiveChannel, customer *CustomerInfo) (err error) {
+	logger := workflow.GetLogger(ctx)
+	activities := Activities{}
+
+	// nothing to receive, but need this to "handle" signal
+	c.Receive(ctx, nil)
+
+	customer.AccountActive = false
+	err = workflow.ExecuteActivity(ctx, activities.SendEmail, common.EmailCancelAccount).Get(ctx, nil)
+	if err != nil {
+		logger.Error("Error running SendEmail activity.", "Error", err)
+		return err
+	}
+
+	logger.Info("Canceled account.", "CustomerID", customer.CustomerId)
+	return nil
+}
+
+func queryGetStatus(ctx workflow.Context, customer CustomerInfo) (GetStatusResponse, error) {
+	logger := workflow.GetLogger(ctx)
+
+	response := GetStatusResponse{
+		Tier:          customer.Status.Tier(),
+		Points:        customer.LoyaltyPoints,
+		AccountActive: customer.AccountActive,
+	}
+	logger.Info("Got response query.", "Customer", customer, "Response", response)
+
+	return response, nil
+}
+
+func queryGetGuests(ctx workflow.Context, customer CustomerInfo) ([]string, error) {
+	logger := workflow.GetLogger(ctx)
+	guestIds := make([]string, len(customer.Guests))
+
+	i := 0
+	for k := range customer.Guests {
+		guestIds[i] = k
+		i++
+	}
```

Comment:
```suggestion
	guestIDs := make([]string, 0, len(customer.Guests))
	for k := range customer.Guests {
		guestIDs = append(guestIDs, k)
	}
```

No need to manually keep track of index, just set capacity of slice and append